src/service/url-replacements-impl.js

```diff
@@ -607,10 +610,12 @@ export class UrlReplacements {
    * @param {!Object<string, *>=} opt_bindings
    * @param {!Object<string, *>=} opt_collectVars
    * @param {boolean=} opt_sync
+   * @param {!Object<string, boolean>=} opt_whiteList Optional white list of names
+   *     that can be substituted.
    * @return {!Promise<string>|string}
    * @private
    */
-  expand_(url, opt_bindings, opt_collectVars, opt_sync) {
+  expand_(url, opt_bindings, opt_collectVars, opt_sync, opt_whiteList) {
```

Comment:
@dvoytenko I think the overall design is moving towards not having any overridden variables. Variables will be either enabled or disabled for `amp-analytics`. This design works for that case(with the change to figure out how whitelist can be converted to blacklist.).